kornia/geometry/liegroup/se3.py

```diff
@@ -63,17 +63,23 @@ def __init__(self, rotation: So3, translation: Tensor) -> None:
         """
         super().__init__()
         KORNIA_CHECK_TYPE(rotation, So3)
+        # KORNIA_CHECK_TYPE(translation, (Vector3, Tensor))
+        assert isinstance(translation, (Vector3, Tensor)), f"translation type is {type(translation)}"
         # KORNIA_CHECK_SHAPE(t, ["B", "3"])  # FIXME: resolve shape bugs. @edgarriba
-        self._rotation = rotation
-        self._translation = Parameter(translation)
+        self._translation: Union[Vector3, Parameter]
+        self._rotation: So3 = rotation
+        if isinstance(translation, Tensor):
+            self._translation = Parameter(translation)
+        else:
+            self._translation = translation
 
     def __repr__(self) -> str:
         return f"rotation: {self.r}\ntranslation: {self.t}"
 
     def __getitem__(self, idx: Union[int, slice]) -> 'Se3':
         return Se3(self._rotation[idx], self._translation[idx])
 
-    def __mul__(self, right: "Se3") -> "Se3":
+    def __mul__(self, right: "Se3") -> Union['Se3', Vector3, Tensor]:
```

Comment:
any case which block us of having `from __future__ import annotations` here too?